manifests/server/config/memory.pp

```diff
@@ -0,0 +1,138 @@
+# clickhouse::server::config::memory
+# This class provide basic memory adjustment for keep ClickHouse server in reasonable limits
+#
+# @param reserved_memory
+#   Memory to leave for system usage and CH overheads like merges and mutations
+#
+# @param mark_cache_size
+#   See CH documentation
+#
+# @param uncompressed_cache_size
+#   See CH documentation
+#
+# @param max_memory_usage_for_all_queries
+#   See CH documentation
+#
+# @param max_memory_usage
+#   See CH documentation
+#
+# @param use_uncompressed_cache
+#   See CH documentation
+#
+# @param external_group_by
+#   See CH documentation
+#
+# @param external_sort
+#   See CH documentation
+#
+# @param memory_check
+#   Check if class has proper parameters and raise errors otherwise
+#
+# @summary Configure memory consumption
+#
+# @see https://clickhouse.tech/docs/en/operations/settings/settings/
+#
+# @see https://clickhouse.tech/docs/en/operations/server_settings/settings/
+#
+# @example Simple use
+#   include clickhouse::server::config::memory
+#
+# @author InnoGames GmbH
+#
+class clickhouse::server::config::memory (
+    Integer[0]           $reserved_memory                  = 4294967296, # 4 GiB for system apps, cache and CH overheads
+    Integer[0]           $mark_cache_size                  = 5368709120, # Must be ≥ 5GiB, direct config parameter
+    Optional[Integer[0]] $uncompressed_cache_size          = undef,
+    Optional[Integer[0]] $max_memory_usage_for_all_queries = undef,
+    Optional[Integer[0]] $max_memory_usage                 = undef,
+    Boolean              $use_uncompressed_cache           = true,
+    Boolean              $external_group_by                = true,
+    Boolean              $external_sort                    = true,
+    Boolean              $memory_check                     = true,
+) inherits clickhouse::server {
+    # Memory calculation
+    if  $mark_cache_size < 5 << 30 and $memory_check {
+        notify { 'Memory':
```

Comment:
lets give more precise resource titles in such a module. other people might want to use `Memory` themselves. i find this too limiting